src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -407,6 +453,10 @@ public static ModelMetadata getMetadataFromSourceMap(final Map<String, Object> m
             methodComponentContext = MethodComponentContext.EMPTY;
         }
 
+        if (vectorDataType == null) {
+            vectorDataType = VectorDataType.FLOAT.getValue();
```

Comment:
I see there are many place where we use float as default value. Can we define default value in VectorDataType and use it instead so that we know where we are using default value in the future?

Reply:
ack